editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/main/MainFragment.kt

```diff
@@ -35,6 +40,14 @@ class MainFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMapLongClickLis
     }
 
     private lateinit var mViewModel: MainViewModel
+    // Terrain that the user is interacting with
+    private var mTerrain: Terrain? = null
+
+    // Tells the main fragment which terrain was associated with the item that was clicked
+    // from TerrainsInListview
+    fun setTerrain(terrain: Terrain) {
```

Comment:
To transfer arguments to a fragment, we can use [`Bundle`s](https://developer.android.com/reference/android/os/Bundle.html)! We'll need to change `Terrain` so it implements [`Parcelable`](https://developer.android.com/reference/android/os/Parcelable.html).

Afterwards, you can make the following code changes to use a bundle:
```diff
class MainFragment {
    ...
    companion object {
-        fun newInstance() = MainFragment()
+        fun newInstance(terrain: Terrain) = MainFragment().apply {
+            arguments = Bundle().apply { putParcelable(TERRAIN_KEY, terrain) }
+        }

+        private const val TERRAIN_KEY = "terrain"
    }
    ...
    override fun onCreateView(...) {
        ...
-        mViewModel.terrain = mTerrain
+        mViewModel.terrain = arguments.getParcelable(TERRAIN_KEY)
        ...
    }
}

class TerrainViewHolder {
    ...
-    Fragment newFragment = new MainFragment();
-    ((MainFragment) newFragment).setTerrain(terrain);
+    Fragment newFragment = MainFragment.newInstance(terrain);
    ...
}
```